lib/heathen/executioner.rb

```diff
@@ -23,7 +25,14 @@ def execute(*argv)
 
       command = argv.map(&:to_s)
 
-      pid, status, @stdout, @stderr = _execute(*command, options)
+      pid, status, @stdout, @stderr =
+        begin
+          Timeout.timeout((ENV['EXECUTE_TIMEOUT'] || '600').to_i) do
```

Comment:
Are you sure timeout cleans the child processes correctly leaving no zombies? I knew `Thread#join(limit)` was the right way to handle this

Reply:
Definitely this would cause zombies. Thank you, I'll rework it to kill the stray processes.